### PR TITLE
TRITON-2001 get 'sdcadm up manta' to create the manta deployment GZ links

### PR DESCRIPTION
**Pull request by trentm**

### CODE REVIEW

File: lib/procedures/ensure-manta-deployment-gz-links.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+function EnsureMantaDeploymentGzLinksProcedure() {
+}
+util.inherits(EnsureMantaDeploymentGzLinksProcedure, Procedure);
+
+EnsureMantaDeploymentGzLinksProcedure.prototype.summarize =
+function summarize() {
+    return 'ensure /opt/smartdc/bin/manta-* tools are setup';
+};
+
+EnsureMantaDeploymentGzLinksProcedure.prototype.execute =
```
Comment (trentm) [inline lib/procedures/ensure-manta-deployment-gz-links.js]: compare to https://github.com/joyent/sdc-headnode/blob/3d07d324fdbd21536315356b47037a7a2d37f7c5/scripts/setup_manta_zone.sh#L178-L241

File: lib/procedures/ensure-manta-deployment-gz-links.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                        'could not read manta-deployment man dir'));
+                    return;
+                }
+                for (let manpage of manpages) {
+                    let link = '/opt/smartdc/man/man1/' + manpage;
+                    let targ = mandir + '/' + manpage;
+                    try {
+                        fs.lstatSync(link);
+                        fs.unlinkSync(link);
+                    } catch (_lstatErr) {
```
Comment (joshwilsdon) [inline lib/procedures/ensure-manta-deployment-gz-links.js]: nit: Is this really only an _lstatErr? Could it not also be an unlink error? E.g. What if someone mistakenly created a directory with the same name as `link` here.

File: lib/procedures/ensure-manta-deployment-gz-links.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                    } else {
+                        ctx.inst = insts[0];
+                        next();
+                    }
+                });
+            },
+
+            // Create stubs in /opt/smartdc/bin/ to a few `manta-*` tools
+            // in the manta-deployment zone.
+            function createLinks(ctx, next) {
```
Comment (joshwilsdon) [inline lib/procedures/ensure-manta-deployment-gz-links.js]: Do we also need to unlink here like the sdc-headnode version does? Or are we guaranteed these are always files (not symlinks or something)?

Comment (trentm) [reply lib/procedures/ensure-manta-deployment-gz-links.js]: Ah, thanks. I'll do as you suggest.

I tested by making one of those a symlink:

```
[root@headnode (coal) /opt/smartdc/bin]# rm manta-login
[root@headnode (coal) /opt/smartdc/bin]# ln -s sdc-role manta-login
[root@headnode (coal) /opt/smartdc/bin]# ls -al manta*
-rwxr-xr-x   1 root     root         211 Nov 27 22:34 manta-adm
lrwxrwxrwx   1 root     root           8 Nov 27 22:36 manta-login -> sdc-role
-rwxr-xr-x   1 root     root         214 Nov 27 22:34 manta-oneach
```

re-running:

```
[root@headnode (coal) /opt/smartdc/bin]# sdcadm up manta --force-same-image
Finding candidate update images for the "manta" service.
Using channel dev

This update will make the following changes:
    update "manta" service to image 27b27f26-dc3c-4308-9abf-64787220c706
        (manta-deployment@master-20191126T202101Z-g764669f):
        instance "91cee159-5e36-4e2c-a310-3badb5f5e53f" (manta0) on server 564d8ce9-08d3-b66d-677f-be7573149cbe
    ensure /opt/smartdc/bin/manta-* tools are setup

Would you like to continue? [y/N] y

Create work dir: /var/sdcadm/updates/20191127T223718Z
Updating image for SAPI service "manta"
    service uuid: 54b99f0e-a568-4a5f-8bf8-64767d445b90
Installing image 27b27f26-dc3c-4308-9abf-64787220c706
    (manta-deployment@master-20191126T202101Z-g764669f)
Reprovisioning VM 91cee159-5e36-4e2c-a310-3badb5f5e53f on server 564d8ce9-08d3-b66d-677f-be7573149cbe
Waiting for manta instance 91cee159-5e36-4e2c-a310-3badb5f5e53f to come up
Wrote "/opt/smartdc/bin/manta-{login,adm,oneach}" stubs
Skipping manta-adm.1 (/opt/smartdc/man/man1/manta-adm.1 already exists)
Creating symlink "/opt/smartdc/man/man1/manta-deploy-coal.1" for "/zones/91cee159-5e36-4e2c-a310-3badb5f5e53f/root/opt/smartdc/manta-deployment/man/man1/manta-deploy-coal.1"
Skipping manta-deploy-dev.1 (/opt/smartdc/man/man1/manta-deploy-dev.1 already exists)
Skipping manta-deploy-lab.1 (/opt/smartdc/man/man1/manta-deploy-lab.1 already exists)
Skipping manta-factoryreset.1 (/opt/smartdc/man/man1/manta-factoryreset.1 already exists)
Skipping manta-init.1 (/opt/smartdc/man/man1/manta-init.1 already exists)
Skipping manta-login.1 (/opt/smartdc/man/man1/manta-login.1 already exists)
Skipping manta-oneach.1 (/opt/smartdc/man/man1/manta-oneach.1 already exists)
Updated successfully (elapsed 34s).
```

and checking:

```
[root@headnode (coal) /opt/smartdc/bin]# ls -al manta*
-rwxr-xr-x   1 root     root         211 Nov 27 22:37 manta-adm
-rwxr-xr-x   1 root     root         115 Nov 27 22:37 manta-login
-rwxr-xr-x   1 root     root         214 Nov 27 22:37 manta-oneach
```

File: lib/procedures/ensure-manta-deployment-gz-links.js (hunk 2 above)
Comment (trentm) [reply lib/procedures/ensure-manta-deployment-gz-links.js]: Updated to skip if the symlink path already exists (as setup_manta_zone.sh does).

**Pull request merged**